### ISSUE TITLE
manual_range_contains misbehaves when bounds have different signs

### ISSUE THREAD

**Issue by xTibor** [rust-lang/rust-clippy#8721]
### Summary

The `manual_range_contains` lint misbehaves when the upper and lower bounds of the range have different signs. There are four test cases in the reproducer:

- Test 1 - Upper and lower bounds are both positive. Suggestions are okay.
- Test 2 - Upper and lower bounds are both negative. Suggestions are okay.
- Test 3 - Upper bound is positive, lower bound is negative. Clippy misses this case, no warnings, no suggestions.
- Test 4 - Upper bound is negative, lower bound is positive. This range is non-sense, however Clippy makes an invalid suggestion here that later triggers an another warning about reversed ranges. Clippy should warn about that non-sense condition instead.

### Reproducer

```rust
fn main() {
    let value = 0;

    // OK suggestion: `(5..=10).contains(&value)`
    let _test_1_pos_pos = (value >= 5) && (value <= 10);

    // OK suggestion: `(-10..=-5).contains(&value)`
    let _test_2_neg_neg = (value >= -10) && (value <= -5);

    // Missing suggestion. Expected: `(-10..=10).contains(&value)`
    let _test_3_neg_pos = (value >= -10) && (value <= 10);

    // Invalid suggestion: `(10..=-10).contains(&value)`
    // The suggested code triggers `clippy::reversed_empty_ranges`.
    let _test_4_pos_neg = (value >= 10) && (value <= -10);
}
```

### Version

```text
rustc 1.62.0-nightly (311e2683e 2022-04-18)
binary: rustc
commit-hash: 311e2683e1bad87715b1558f7900e294d24ce491
commit-date: 2022-04-18
host: x86_64-unknown-linux-gnu
release: 1.62.0-nightly
LLVM version: 14.0.1
```
